http2/src/main/java/com/blazemeter/jmeter/http2/sampler/HTTP2SampleResult.java

```diff
@@ -1,432 +1,427 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.blazemeter.jmeter.http2.sampler;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.thoughtworks.xstream.XStream;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import org.apache.jmeter.protocol.http.sampler.HTTPSampleResult;
+import org.apache.jmeter.protocol.http.util.HTTPConstants;
+import org.apache.jmeter.samplers.SampleEvent;
+import org.apache.jmeter.samplers.SampleResult;
+import org.apache.jmeter.save.SaveService;
+import org.apache.jmeter.threads.JMeterThread;
+import org.apache.jmeter.threads.JMeterVariables;
+import org.apache.jmeter.threads.ListenerNotifier;
+import org.apache.jmeter.threads.SamplePackage;
+import org.eclipse.jetty.http.HttpFields;
+
+/**
+ * This is a specialisation of the SampleResult class for the HTTP protocol.
+ */
+public class HTTP2SampleResult extends HTTPSampleResult {
+
+  private static final long serialVersionUID = 241L;
+
+  /**
+   * Set of all HTTP methods, that have no body
+   */
+  private static final Set<String> METHODS_WITHOUT_BODY = new HashSet<>(
+      Arrays.asList(HTTPConstants.HEAD, HTTPConstants.OPTIONS, HTTPConstants.TRACE));
+
+  private static int idCount = 0;
+
+  private int id;
+
+  private int embebedResultsDepth;
+  private HttpFields httpFieldsResponse;
+  private boolean embebedResults;
+  private transient Queue<HTTP2SampleResult> pendingResults = new ConcurrentLinkedQueue<>();
+
+  private boolean secondaryRequest;
+
+  private String embeddedUrlRE;
+
+  private boolean isPushed;
+
+  private transient ListenerNotifier listenerNotifier = new ListenerNotifier();
+
+  private transient JMeterVariables threadVars;
+  private transient SamplePackage pack;
+  /**
+   * The raw value of the Location: header; may be null. This is supposed to be an absolute URL: <a
+   * href= "http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.30">RFC2616 sec14.30</a>
+   * but is often relative.
+   */
+  private String redirectLocation;
+
+  protected static final String NON_HTTP_RESPONSE_CODE = "Non HTTP response code";
+
+  protected static final String NON_HTTP_RESPONSE_MESSAGE = "Non HTTP response message";
+  protected static final String HTTP2_PENDING_RESPONSE = "Pending";
+  protected static final String HTTP2_RESPONSE_RECEIVED = "Received";
+  protected static final String HTTP2_RESPONSE_CODE_4 = "Not Found";
+  private boolean pendingResponse;
+
+  private String requestId;
+
+  static {
+    try {

```

Comment:
create a static method for this logic with proper name (as a way to document it) and invoke the method from this static block. Maybe something like registerHttp2ResultConverter.